src/cursor/abstract_cursor.ts

```diff
@@ -284,11 +286,40 @@ export abstract class AbstractCursor<
   }
 
   [Symbol.asyncIterator](): AsyncIterator<TSchema, void> {
+    async function* nativeAsyncIterator(this: AbstractCursor<TSchema>) {
```

Comment:
Is this necessary?  I understand the desire to use generators wherever possible, but wouldn't it be simpler to just leave the implementation as it was (perhaps adjusting for closed cursors)?

Reply:
> Implement the async iterator using native syntax when a custom promise is not set. I think this is important to do now ensuring is maintains the expected behavior so when we drop custom promise support we don't miss something in the other API removals/changes that would be part of the major.

I think it's worth doing now, but I think we can better reuse the code, take another look, I use the same iterator, just wrap the promises if needed

Reply:
So the idea is, once we're in v5, so rewrite it as a standalone async generator?  

```typescript
[Symbol.asyncIterator] = async function*() { .... } 
```
?

Reply:
Pretty much more like: 
```ts
class {
async *[Symbol.asyncIterator]() {
  // contents of the iterator we've declared inline for v4
}
}
```
so the function will be directly on the class instead of this custom promise wrapping. The idea is to not do the done/value stuff ourselves, it's built in to the language if you just use the syntax.